ftw/linkchecker/tests/test_functionality.py

```diff
@@ -0,0 +1,145 @@
+from AccessControl.SecurityManagement import newSecurityManager
+from ftw.builder import Builder
+from ftw.builder import create
+from ftw.linkchecker import report_generating
+from ftw.linkchecker import report_mailer
+from ftw.linkchecker.cell_format import BOLD
+from ftw.linkchecker.cell_format import CENTER
+from ftw.linkchecker.cell_format import DEFAULT_FONTNAME
+from ftw.linkchecker.cell_format import DEFAULT_FONTSIZE
+from ftw.linkchecker.tests.base import FunctionalTestCase
+from ftw.testbrowser import browsing
+from ftw.testing.mailing import Mailing
+from zope.component.hooks import setSite
+import AccessControl
+import email
+import os
+import pandas as pd
+import transaction
+
+CURRENT_PATH = os.path.dirname(os.path.abspath(__file__))
+
+
+class TestLinkChecker(FunctionalTestCase):
+
+    @browsing
+    def test_finds_links_in_plone_site(self, browsing):
+        """Checks if links in page are found and broken ones can be excluded.
+        """
+        self.set_up_test_environment()
+        pass
+
+    @browsing
+    def test_finds_broken_relations(self, browsing):
+        """Checks if broken relations in page are found.
+        """
+        self.set_up_test_environment()
+        pass
+
+    def test_if_excel_generator_adds_content_correctly(self):
+        """Test if an excel workbook generated by the linkchecker does not
+        differ an exemplar workbook containing the expected data.
+        """
+        path_of_excel_workbook_generated = (
+                CURRENT_PATH +
+                '/exemplar_data/generator_excel_sheet_to_compare.xlsx')
+        path_of_excel_workbook_exemplar = (
+                CURRENT_PATH +
+                '/exemplar_data/expected_excel_sheet_outcome.xlsx')
+
+        self.generate_test_data_excel_workbook(
+            path_of_excel_workbook_generated)
+
+        # import the excel workbooks as pandas dataframes
+        df1 = pd.read_excel(
+            path_of_excel_workbook_generated)
+        df2 = pd.read_excel(
+            path_of_excel_workbook_exemplar)
+
+        assert df1.equals(df2), \
+            "The examplar excel workbook converges from the one generated."
+
+    def test_if_mail_sender_sending_mail_incl_attachement(self):
+        """Test if the mail sent by linkchecker can be received correctly.
+        """
+
+        # test email variables
+        email_subject = 'Linkchecker Report'
+        email_message = 'Dear Site Administrator, \n\n\
+            Please check out the linkchecker report attached to this mail.\n\n\
+            Friendly regards,\n\
+            your 4teamwork linkcheck reporter'
+        receiver_email_address = 'hugo@boss.ch'
+        report_path = CURRENT_PATH + \
+                      '/exemplar_data/expected_excel_sheet_outcome.xlsx'
+
+        # setUp
+        Mailing(self.layer['portal']).set_up()
+        transaction.commit()
```

Comment:
Since there's no other thread making requests, you don't have to commit those changes.